Functions/Gherkin.Core.ps1

```diff
@@ -0,0 +1,1454 @@
+        $StepFiles = & $SafeCommands['Get-ChildItem'] $StepPath -Filter "*.?teps.ps1" -Include "*.[sS]teps.ps1" -Recurse
+
+        foreach ($StepFile in $StepFiles) {
+            $invokeTestScript = {
+                [CmdletBinding()]
+                param (
+                    [Parameter(Position = 0)]
+                    [string] $Path
+                )
+
+                & $Path
+            }
+
+            Set-ScriptBlockScope -ScriptBlock $invokeTestScript -SessionState $Pester.SessionState
+
+            & $invokeTestScript $StepFile.FullName
+        }
+
+        & $SafeCommands['Write-Verbose'] "Loaded $($Script:GherkinStepDefinitions.Count) step definitions from $(@($StepFiles).Count) steps file(s)"
+    }
+}
+
+function Import-GherkinFeature {
+    <#
+        .SYNOPSIS
+            Internal function to import a Gherkin feature file. Wraps Gherkin.Parse
+
+        .PARAMETER Path
+            The path to the feature file to import
+
+        .PARAMETER Pester
+            Internal Pester object. For internal use only
+
+        .PARAMETER NoMultiline
+            Controls whether or not StepDefinition DocStrings and DataTables are printed to the console during
+            the test run.
+
+        .PARAMETER Expand
+            Controls whether or not Scenario Outline example scenarios are displayed fully expanded instead of
+            just printing the secnario outline and scenario example table. This can be useful to determine
+            exactly which step of a scenario outline example scenario failed.
+    #>
+    [CmdletBinding()]
+    param(
+        [Parameter(Position = 0, Mandatory = $True)]
+        [string]$Path,
+
+        [Parameter(Position = 1, Mandatory = $True)]
+        [PSObject]$Pester,
+
+        [switch]$NoMultiline,
+
+        [switch]$Expand
+    )
+
+    $AddMember     = $SafeCommands['Add-Member']
+    $CompareObject = $SafeCommands['Compare-Object']
+    $GetUnique     = $SafeCommands['Get-Unique']
+    $MeasureObject = $SafeCommands['Measure-Object']
+    $NewObject     = $SafeCommands['New-Object']
+    $SelectObject  = $SafeCommands['Select-Object']
+    $WhereObject   = $SafeCommands['Where-Object']
+    $WriteWarning  = $SafeCommands['Write-Warning']
+
+    $Background = $null
+
+    $Parser = & $NewObject Gherkin.Parser
+    $Feature = $Parser.Parse($Path).Feature | Convert-Tags
+
+    # If the entire feature is excluded by tags, then don't process it any further.
+    if ($Pester.ExcludeTagFilter -and @(& $CompareObject $Feature.Tags $Pester.ExcludeTagFilter -IncludeEqual -ExcludeDifferent).Length -gt 0) {
+        return
+    }
+
+    $Scenarios = @(
+        :scenarios foreach ($Child in $Feature.Children) {
+            $null = & $AddMember -MemberType NoteProperty -InputObject $Child.Location -Name Path -Value $Path
+
+            foreach ($Step in $Child.Steps) {
+                $null = & $AddMember -MemberType NoteProperty -InputObject $Step.Location -Name Path -Value $Path
```

Comment:
Apparently, this line, and line 493 above, never worked in PowerShell v2.0. The Path property is never added to the ETS on the Location object referenced by the Location property of the respective objects. This is because the Location property on those objects is a read-only property (it's declared as `Gherkin.Ast.Location Location { get; private set; }`). There must have been some compensating code (very far removed) somewhere to make the stack trace nice, for example. Anyway, these lines are causing 4 unit test failures. I'm evaluating how to make this work in PowerShell v2.

Reply:
Just to provide a bit more detail, I ran this code in PowerShell v5.x. In the Debugger windows, the Steps (for example) did not show a `Path` property on the Location property either. _However_, when performing a `$Child.Steps[0].Location | Get-Member -View Extended` call, the `Path` property was returned. So it _is_ available--which makes sense since the tests do pass on PowerShell v3 and above. And this is expected since the way the ETS works changed between PSv2 and PSv3.

Reply:
OK, there was some (far removed) compensating code that handled this PowerShell v2 behavior. Instead of having 2 ways to deal with this, I used a PSv2 compatible approach. This will be fixed in the next set of commits.